PetrovAI/lab_3/main.cpp

```diff
@@ -0,0 +1,207 @@
+#include <iostream>
```

Comment:
You need to add destructors to both d_array and list. Currently your classes causes memory leak